src/greeter/XcbKeyboardBackend.cpp

```diff
@@ -80,7 +80,7 @@ namespace SDDM {
         xcb_generic_error_t *error = nullptr;
 
         m_conn = xcb_connect(nullptr, nullptr);
-        if (m_conn == nullptr) {
+        if (xcb_connection_has_error(m_conn)) {
             qCritical() << "xcb_connect failed, keyboard extension disabled";
```

Comment:
In that case it would also have to `xcb_disconnect(m_conn); m_conn = nullptr;` for cleanup.

Reply:
your suggestion makes perfect sense, but I'm on vacation for a few days. I was going to work on it today, but you've already merged it.